src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -130,6 +131,120 @@ internal ProcessOutputObject(object data, MinishellStream stream)
         }
     }
 
+    /// <summary>
+    /// This exception is used by the NativeCommandProcessor to indicate an error
+    /// when a native command retuns a non-zero exit code.
+    /// </summary>
+    [Serializable]
+    public class NativeCommandExitException : RuntimeException
+    {
+        // NOTE:
+        // When implementing the native error action preference integration,
+        // reusing ApplicationFailedException was contemplated.
+        // However, rather than possibly reusing a type already used in another scenario
+        // it was decided instead to use a fresh type to minimize the chance of conflating the two scenarios.
```

Comment:
```suggestion
        // NOTE:
        // When implementing the native error action preference integration,
        // reusing ApplicationFailedException was rejected.
        // Instead of reusing the type already used in another scenario
        // it was decided instead to use a fresh type to minimize the chance of conflating the two scenarios:
        // - ApplicationFailedException - PowerShell can not execute an application.
        // - NativeCommandExitException - an application was successfully executed and completed but return non-zero  code.
```